library/control/src/modules/WorkflowManager.rb

```diff
@@ -477,25 +495,24 @@ def GetCachedWorkflowFilename(type, src_id, _name)
 
           # The most generic way it to use the package referenced by the "installerextension()"
           # provides, this works with all repository types, including the RPM-MD repositories.
-          use_filename ||= addon_control_file(src_id)
-
-          # File exists?
-          return use_filename.nil? ? nil : StoreWorkflowFile(use_filename, disk_filename)
+          use_filename ||= control_file(src_id)
+        when :package
+          use_filename = control_file(name)
+        else
+          raise ArgumentError, "invalid argument type #{type.inspect}"
```

Comment:
This default branch should be actually never called, there is an argument check at the beginning. So either remove it or improve the error message so it's clear that the previous check is broken.